### PR TITLE
Style mapping

### PR DESCRIPTION
**Pull request by b4l**
Depends on #2

### CODE REVIEW

File: src/MapfileStyleParser.ts [hunk 1]
```diff
@@ -28,20 +37,331 @@ export class MapfileStyleParser implements StyleParser {
...
   }
 
+  /**
+   * Get the name for the Style from the Mapfile Object. Returns the GROUP value of the LAYER
+   * if defined or the NAME value of the LAYER if defined or an empty string.
+   *
+   * @param {object} mapfileObject The Mapfile object representation (created with xml2js)
+   * @return {string} The name to be used for the GeoStyler Style Style
+   */
+  getStyleNameFromMapfileObject(mapfileObject: any): string {
```
Comment (bchapuis) [inline src/MapfileStyleParser.ts:47]: Could we introduce a type for mapfileObject in the parser? or is it too convoluted?

File: src/MapfileStyleParser.ts [hunk 2]
```diff
@@ -28,20 +37,331 @@ export class MapfileStyleParser implements StyleParser {
...
+  }
+
+
+  /**
+   * Get the GeoStyler-Style Filter from an Mapfile Expression.
+   *
+   * @param {string} mapfileExpression The Mapfile Expression
+   * @return {Filter} The GeoStyler-Style Filter
+   */
+  getFilterFromExpression(mapfileExpression: any): Filter | undefined {
```
Comment (bchapuis) [inline src/MapfileStyleParser.ts:60]: Same here and after (: any)?

File: src/mapfile2js/src/parse/parseBlockKey.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * @param {Object} lo Line Object
+ * @param {number} index Current lines index
+ * @param {array} lines Array of line strings
+ * @param {array} blocks Block stack
+ */
+
+// there can be multiple layer, class and style block siblings
+const multi_block_keys = ["layer", "class", "style"];
+
+function parseBlockKey(lo, index, lines, blocks) {
```
Comment (bchapuis) [inline src/mapfile2js/src/parse/parseBlockKey.js:12]: Can we avoid side effects on the blocks variable?

Comment (b4l) [reply src/mapfile2js/src/parse/parseBlockKey.js]: yes, done in #2

File: src/MapfileStyleParser.ts (hunk 2 above)
Comment (b4l) [reply src/MapfileStyleParser.ts]: changed to string

File: src/MapfileStyleParser.ts (hunk 1 above)
Comment (b4l) [reply src/MapfileStyleParser.ts:44]: leaving it as is for now

**Pull request merged**